### PR TITLE
Minor Py3 related tests fixes.

### PR DESCRIPTION
**Pull request by DmitryKuzmenko**
### What does this PR do?
Here are some Py3 tests related fixes that I've made by testing on my Arch dev box.

### What issues does this PR fix or reference?
Related to saltstack/salt-jenkins#333

### CODE REVIEW

File: salt/utils/boto.py [hunk 1]
```diff
@@ -179,8 +179,10 @@ def get_connection(service, module=None, region=None, key=None, keyid=None,
     '''
 
     module = module or service
+    module = 'boto.' + module
+    module, submodule = module.rsplit('.', 1)
```
Comment (isbm) [inline salt/utils/boto.py]: I am a bit confused, why is this needed? As easy as:
```python
module, submodule = 'boto', module
```

File: tests/unit/pillar/test_mysql.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
 
 
 @skipIf(NO_MOCK, NO_MOCK_REASON)
-@skipIf(PY3, 'MySQL-python is not compatible with python3')
+@skipIf(mysql.HAS_MYSQL is False, 'Install MySQL bindings before running MySQL unit tests.')
```
Comment (isbm) [inline tests/unit/pillar/test_mysql.py]: It is easier to read in English "not has mysql" (understood as "has no mysql") instead of "has mysql is false" (understood as "the statement of having mysql is wrong"). And is shorter. So:
```python
@skipif(not mysql.HAS_MYSQL, 'Install.......')
```

**Review by isbm — CHANGES_REQUESTED** [review]

File: salt/utils/boto.py (hunk 1 above)
Comment (DmitryKuzmenko) [inline salt/utils/boto.py]: We need it for the case if module contains `.`:
```
module = 'module.submodule'
module = 'boto.' + module => 'boto.module.submodule'
module, submodule = module.rsplit('.', 1) => 'boto.module', 'submodule'

File: tests/unit/pillar/test_mysql.py (hunk 2 above)
Comment (DmitryKuzmenko) [inline tests/unit/pillar/test_mysql.py]: @isbm agree with you. I've wrote it in that way just because of this: 😄 
```
$ grep -Rn "skipIf.*HAS_" tests/ | grep "is False" | wc -l
135
$ grep -Rn "skipIf.*not.*HAS_" tests/ | wc -l
78
```
Anyway, I'll update that.
Thank you for review.

File: salt/utils/boto.py (hunk 1 above)
Comment (isbm) [inline salt/utils/boto.py]: Ah. Well OK, but in this case if `module` contains sub-submodule like `module.submodule.subsubmodule` you have a problem. Additionally, concatenating strings is not the very best style. I would still write it this way (oneliners rules, LOL):
```python
module, submodule = (lambda mod ['.'.join(pk) for pk in [mod[:2], mod[2:]]])(['boto'] + module.split('.'))
```

Comment (DmitryKuzmenko) [inline salt/utils/boto.py]: Sorry, as I see your code does exactly the same as mine, and your version looks more Pythonic, but it slower and less readable. Probably you're right about the triple `module` assigning. I've rewrote the last two lines to a single one.
```bash
$ python -m timeit -s "module = 'a.b.c.d'" "module, submodule = (lambda mod: ['.'.join(pk) for pk in [mod[:-1], mod[-1:]]])(['boto'] + module.split('.'))"
1000000 loops, best of 3: 1.12 usec per loop
$ python -m timeit -s "module = 'a.b.c.d'" "module = 'boto.' + module; module, submodule = module.rsplit('.', 1)"
1000000 loops, best of 3: 0.285 usec per loop
$ python -m timeit -s "module = 'a.b.c.d'" "module, submodule = ('boto.' + module).rsplit('.', 1)"
1000000 loops, best of 3: 0.281 usec per loop
```
